bundle/Controller/Admin/TagController.php

```diff
@@ -500,9 +500,49 @@ public function childrenAction(Request $request, ?Tag $tag = null): Response
             );
         }
 
+        if ($request->request->has('HideTagsAction')) {
+            return $this->redirectToRoute(
+                'netgen_tags_admin_tag_hide_tags',
+                [
+                    'parentId' => $tag?->id ?? 0,
+                ],
+            );
+        }
+
+        if ($request->request->has('RevealTagsAction')) {
+            return $this->redirectToRoute(
+                'netgen_tags_admin_tag_reveal_tags',
+                [
+                    'parentId' => $tag?->id ?? 0,
+                ],
+            );
+        }
+
         return $this->redirect($request->getPathInfo());
     }
 
+    public function hideAction(Request $request, Tag $tag): Response
+    {
+        $this->denyAccessUnlessGranted('ibexa:tags:hide' . ($tag->isSynonym() ? 'synonym' : ''));
+
+        $this->tagsService->hideTag($tag);
```

Comment:
Do we need to check the submitted button and CSRF token here?

Reply:
Yes, we do. There is an example in `deleteTagAction` method.